pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -402,6 +397,15 @@ def _apply_lightning_controlled(self, sim, operation):
                         operation.base.matrix, control_wires, control_values, target_wires, False
                     )
 
+        def supports_operation(self, operation):
+            """Overwrite base class implementation to allow arbitrarily nested Controlled instances
+            to be applied by Lightning."""
+
+            while operation[0:2] == "C(":
+                operation = operation[2:-1]
+
+            return super().supports_operation(operation)
```

Comment:
@dime10 could you please clarify how e.g. `C(CX)` gate would be handled? From my understanding, Lighting does support `CX` but does not natively support the controlled version of this gate.

Reply:
It seems PennyLane will spit out `C(CX)` as a `Toffoli`, if it has a single control, and `MultiControlledX` otherwise. This is handled by the Python module and the C++ layer. The latter treats CNOT and Toffoli gates separately, but all `MultiControlledX` are understood as `C(PauliX)`, which are natively supported.

Reply:
A slight caveat here: a `qml.ctrl(op, wire)` where `op` is a `CNOT` will spit out `Toffoli`, but if the operation is instantiated using the constructor of `Controlled` like `Controlled(op, wire)`, then it doesn't magically becomes a `Toffoli`, in which case you will need to call `op.simplify()` to flatten the nested control structure.